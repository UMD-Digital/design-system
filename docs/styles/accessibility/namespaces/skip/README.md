[**@universityofmaryland/web-styles-library**](../../../README.md)

***

[@universityofmaryland/web-styles-library](../../../README.md) / [accessibility](../../README.md) / skip

# skip

## Variables

- [content](variables/content.md)
