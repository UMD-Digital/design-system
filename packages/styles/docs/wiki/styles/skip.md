[umd-styles-library](../wiki/Home) / skip

# skip

## Variables

| Variable | Description |
| ------ | ------ |
| [content](../wiki/skip.Variable.content) | Skip to content link class. |
