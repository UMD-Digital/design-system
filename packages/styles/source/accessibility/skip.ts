import { create } from '../utilities';
import type { JssObject } from '../_types';

/**
 * Skip to content link class.
 * @returns {JssObject} The JSS object for the skip to content link.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.accessibility.skip.content
 * ```
 *
 * @since 1.8.0
 */
export const content: JssObject = create.jssObject({
  className: 'umd-skip-content',
  backgroundColor: `#fff`,
  color: `#e21833`,
  display: `block`,
  height: `0`,
  width: '0',
  opacity: `0`,
  padding: `8px`,
  position: `absolute`,
  textDecoration: `underline`,
  transition: `none`,
  overflow: 'hidden',
  whiteSpace: 'nowrap',

  '&:focus': {
    display: 'block',
    height: 'inherit',
    width: 'auto',
    zIndex: '9999',
    opacity: '1',
    transition: 'opacity 0.5s',
  },
});
