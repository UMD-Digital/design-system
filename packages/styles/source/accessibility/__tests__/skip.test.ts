import { content } from '../skip';

describe('accessibility/skip', () => {
  describe('content', () => {
    it('should match snapshot', () => {
      expect(content).toMatchSnapshot();
    });

    it('should have correct className', () => {
      expect(content.className).toBe('umd-skip-content');
    });

    it('should have appropriate base styles', () => {
      expect(content.backgroundColor).toBe('#fff');
      expect(content.color).toBe('#e21833');
      expect(content.position).toBe('absolute');
      expect(content.opacity).toBe('0');
      expect(content.overflow).toBe('hidden');
      expect(content.whiteSpace).toBe('nowrap');
    });

    it('should have appropriate focus styles', () => {
      const focusStyles = content['&:focus'];
      expect(focusStyles).toBeDefined();
      expect(focusStyles.display).toBe('block');
      expect(focusStyles.opacity).toBe('1');
      expect(focusStyles.zIndex).toBe('9999');
    });
  });
});