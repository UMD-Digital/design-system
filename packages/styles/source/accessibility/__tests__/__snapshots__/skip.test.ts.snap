// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accessibility/skip content should match snapshot 1`] = `
{
  "&:focus": {
    "display": "block",
    "height": "inherit",
    "opacity": "1",
    "transition": "opacity 0.5s",
    "width": "auto",
    "zIndex": "9999",
  },
  "backgroundColor": "#fff",
  "className": "umd-skip-content",
  "color": "#e21833",
  "display": "block",
  "height": "0",
  "opacity": "0",
  "overflow": "hidden",
  "padding": "8px",
  "position": "absolute",
  "textDecoration": "underline",
  "transition": "none",
  "whiteSpace": "nowrap",
  "width": "0",
}
`;
